numpy/lib/_shape_base_impl.py

```diff
@@ -1260,17 +1276,32 @@ def tile(A, reps):
         tup = tuple(reps)
     except TypeError:
         tup = (reps,)
-    d = len(tup)
     if all(x == 1 for x in tup) and isinstance(A, _nx.ndarray):
         # Fixes the problem that the function does not make a copy if A is a
         # numpy array and the repetitions are 1 in all dimensions
-        return _nx.array(A, copy=True, subok=True, ndmin=d)
+        return _nx.array(A, copy=True, subok=True, ndmin=len(tup))
     else:
         # Note that no copy of zero-sized arrays is made. However since they
         # have no data there is no risk of an inadvertent overwrite.
-        c = _nx.array(A, copy=False, subok=True, ndmin=d)
-    if (d < c.ndim):
-        tup = (1,)*(c.ndim-d) + tup
+        c = _nx.array(A, copy=False, subok=True, ndmin=len(tup))
+
+    if isinstance(axis, int):
```

Comment:
Why not just make `axis` a tuple and compare lengths after? I.e., something like (keeping the `d` from above - do feel free to rename it...):
```
if axis is None:
    axis = tuple(range(-d, 0))
else:
    if not isinstance(axis, tuple):
        axis = (axis,)
    elif len(set(axis)) != len(axis):
        raise ValueError("duplicates in `axis`.")
    if len(axis) != d:
        raise ValueError("`reps` and `axis` should have the same length")

tup = tuple((tup[axis.index(i)] if i in axis else 1) for i in range(c.ndim))
shape_out = ...
```
(note: I didn't test this but hopefully it is correct...)